app/pages/ServiceDiscoveryForm/ServiceDiscoveryForm.tsx

```diff
@@ -4,63 +4,117 @@ import qs from 'qs';
   interface MatchParams {
     categorySlug: string;
   }
 
+  const match = useRouteMatch();
   const { categorySlug } = match.params as MatchParams;
   const category = CATEGORIES.find(c => c.slug === categorySlug);
   if (!category) {
     // Category does not exist; user may have entered the category in the URL bar
-    // or there is an error in our code
     return <Redirect push to={{ pathname: '/' }} />;
   }
 
-  const eligibilities: CategoryRefinement[] = useEligibilitiesForCategory(category.id) || [];
-  const subcategories: CategoryRefinement[] = useSubcategoriesForCategory(category.id) || [];
+  // The activeCategoryId is updated if the user proceeds to a further step that has child
+  // subcategories to be displayed. When it is set, the target subcategory refinements are
+  // fetched and rendered in place of the previous refinements
+  const [activeCategoryId, setActiveCategoryId] = useState<string | null>(category.id);
+
+  // Add the selectedRadioItem to state so that when a user clicks the next button
+  // the selectedRadioItem can be passed to the goToNextStep function
+  const [selectedRadioItem, setSelectedRadioItem] = useState<number | null>(null);
+
+  const eligibilities: CategoryRefinement[] = useEligibilitiesForCategory(activeCategoryId)
+  || [];
+  const subcategories: CategoryRefinement[] = useSubcategoriesForCategory(activeCategoryId)
+  || [];
+
   return (
     <InnerServiceDiscoveryForm
-      categorySlug={category.slug}
+      category={category}
+      subcategorySubheading={category.subcategorySubheading}
       eligibilities={eligibilities}
       subcategories={subcategories}
-      steps={category.steps}
-      subcategorySubheading={category.subcategorySubheading}
+      setActiveCategoryId={setActiveCategoryId}
+      selectedRadioItem={selectedRadioItem}
+      setSelectedRadioItem={setSelectedRadioItem}
     />
   );
 };
 
 /** Main component that handles form data and advancing steps. */
 const InnerServiceDiscoveryForm = ({
-  steps, eligibilities, subcategories, categorySlug, subcategorySubheading,
+  category, subcategorySubheading, eligibilities, subcategories,
+  setActiveCategoryId, selectedRadioItem, setSelectedRadioItem,
 }: {
-  steps: Step[];
+  category: ServiceCategory;
+  subcategorySubheading: string;
   eligibilities: CategoryRefinement[];
   subcategories: CategoryRefinement[];
-  categorySlug: string;
-  subcategorySubheading: string;
+  setActiveCategoryId: (id: string | null) => void;
+  selectedRadioItem: number | null;
+  setSelectedRadioItem: (item: number | null) => void;
 }) => {
-  const [currentStep, setCurrentStep] = useState(0);
+  const {
+    steps,
+    slug: categorySlug,
+  } = category;
 
-  // TODO: Should goBack go back to the previous step?
+  const [currentStep, setCurrentStep] = useState(0);
   const history = useHistory();
-  const goBack = () => history.goBack();
-  const goToNextStep = () => setCurrentStep(currentStep + 1);
+  let goToNextStep;
```

Comment:
Apologies if I made this comment in some older PR, but the code might be a bit easier to follow if moved the `if (stepName === 'housingStatus')` conditional into the definition of `goToNextStep` rather than conditionally assigning two different functions. E.g.:

```js
const goToNextStep = () => {
  if (stepName === 'housingStatus') {
    // ...
  } else {
    // ...
  }
}
```

Reply:
Oh yes, excellent suggestion. Thank you!